### PR TITLE
Add support for Python 3.10, drop EOL 3.5, replace Travis CI with GitHub Actions

### PR DESCRIPTION
**Pull request by hugovk**
* Add support for Python 3.10, due out in a couple of weeks: https://www.python.org/dev/peps/pep-0619/
* Drop support for EOL Python 3.5.
* Travis CI is being turned off for the pytest-dev org ([internal discussion](https://github.com/orgs/pytest-dev/teams/contributors/discussions/5)), due to the recent security handling ([1](https://twitter.com/peter_szilagyi/status/1437646118700175360), [2](https://arstechnica.com/information-technology/2021/09/travis-ci-flaw-exposed-secrets-for-thousands-of-open-source-projects/)), so replace CI with GitHub Actions
* This lets us also easily test on macOS and Windows!
* Also update some URLs from old https://github.com/ImXron/pytest-stress to new https://github.com/pytest-dev/pytest-stress

### CODE REVIEW

**Comment by hugovk** [comment]
Example build:

https://github.com/hugovk/pytest-stress/actions/runs/1256940560

**Review by ImXron — APPROVED** [review]
Thanks for doing this! Feel free to squash/merge when you're ready.

Do we also need to remove Travis from this package's settings?

**Comment by hugovk** [comment]
> Thanks for doing this! Feel free to squash/merge when you're ready.

You're welcome! I don't have merge rights here, but I'm otherwise ready.
 
> Do we also need to remove Travis from this package's settings?

If you mean these required status checks, yes, these'll need removing (and optionally new GHA ones adding after merge):

image

If you mean disabling Travis from this repo, no need, @RonnyPfannschmidt will be disabling it from the whole pytest-dev org at some point:

* https://github.com/orgs/pytest-dev/teams/contributors/discussions/5

**Pull request merged**
